src/providers/mdal/qgsmdalprovider.cpp

```diff
@@ -241,6 +234,49 @@ bool QgsMdalProvider::persistDatasetGroup( const QString &path,
   return false;
 }
 
+void QgsMdalProvider::loadData()
+{
+  noExtraDatasetGroupsCount = 0;
+  QByteArray curi = dataSourceUri().toUtf8();
+  mMeshH = MDAL_LoadMesh( curi.constData() );
+  if ( mMeshH )
+  {
+    const QString proj = MDAL_M_projection( mMeshH );
+    if ( !proj.isEmpty() )
+      mCrs.createFromString( proj );
+
+    noExtraDatasetGroupsCount = MDAL_M_datasetGroupCount( mMeshH );
+  }
+}
+
+void QgsMdalProvider::reloadData()
+{
+  if ( mMeshH )
+    MDAL_CloseMesh( mMeshH );
+
+  loadData();
+
+  if ( mMeshH )
+  {
+    bool atLeastOneExtraNotLoaded = false;;
+    for ( auto uri : mExtraDatasetUris )
+    {
+      std::string str = uri.toStdString();
+      MDAL_M_LoadDatasets( mMeshH, str.c_str() );
+
+      if ( MDAL_LastStatus() != MDAL_Status::None )
+      {
+        atLeastOneExtraNotLoaded |= true;
```

Comment:
why not `atLeastOneExtraNotLoaded=true`

Reply:
Hi,

I have purposed this logic to avoid removing all the users renderer settings, to reduce this dommageable consequences. I supposed the count of dataset groups present in the mesh file couldn't change, and, in fact, maybe it could change as well... 
So, in the new commit i have just purposed, if  extras uris can't be loaded **or** dataset groups count of the mesh file has changed, all the renderer settings is removed.
It is a bit confusing and I think there still are some cases that are not cover.
Maybe not satisfying...
I don't have time anymore today to think about, I will do later.